runtime/plaid/src/storage/dynamodb/mod.rs

```diff
@@ -0,0 +1,297 @@
+
+        let items = dynamodb_query(&self.client, &self.table_name, namespace, prefix, KEY).await?;
+
+        // Add retrieved items to our growing list
+        for item in &items {
+            all_keys.push(
+                item.get(KEY)
+                    .ok_or(StorageError::Access(
+                        "Could not list storage contents".to_string(),
+                    ))?
+                    .as_s()
+                    .map_err(|_| {
+                        StorageError::Access("Could not list storage contents".to_string())
+                    })?
+                    .clone(),
+            );
+        }
+
+        Ok(all_keys)
+    }
+
+    async fn fetch_all(
+        &self,
+        namespace: &str,
+        prefix: Option<&str>,
+    ) -> Result<Vec<(String, Vec<u8>)>, StorageError> {
+        let mut everything = vec![];
+
+        let items = dynamodb_query(
+            &self.client,
+            &self.table_name,
+            namespace,
+            prefix,
+            "key,value",
+        )
+        .await?;
+
+        // Add retrieved items to our growing list
+        for item in &items {
+            everything.push((
+                item.get(KEY)
+                    .ok_or(StorageError::Access(
+                        "Could not list storage contents".to_string(),
+                    ))?
+                    .as_s()
+                    .map_err(|_| {
+                        StorageError::Access("Could not list storage contents".to_string())
+                    })?
+                    .clone(),
+                item.get(VALUE)
+                    .ok_or(StorageError::Access(
+                        "Could not list storage contents".to_string(),
+                    ))?
+                    .as_b()
+                    .map_err(|_| {
+                        StorageError::Access("Could not list storage contents".to_string())
+                    })?
+                    .clone()
+                    .into_inner(),
+            ));
+        }
+
+        Ok(everything)
+    }
+
+    async fn get_namespace_byte_size(&self, namespace: &str) -> Result<u64, StorageError> {
+        let all = self.fetch_all(namespace, None).await?;
+        let mut counter = 0u64;
+        for item in all {
+            // Count bytes for keys and values
+            counter = counter + item.0.as_bytes().len() as u64 + item.1.len() as u64;
+        }
+        Ok(counter)
+    }
+}
+
+/// Perform a query on DynamoDB.
+///
+/// Args:  
+/// `client` - a DynamoDB Client  
+/// `table_name` - the name of the DynamoDB table  
+/// `namespace` - the namespace we want to query  
+/// `prefix` - [Optional] a prefix that keys must have in order to be returned by this query  
+/// `attributes_to_get` - A comma-separated list of attributes to get from DynamoDB
+async fn dynamodb_query(
+    client: &Client,
+    table_name: &str,
+    namespace: &str,
+    prefix: Option<&str>,
+    attributes_to_get: &str,
```

Comment:
I think this should be a `Vec` then turned into a str inside this function. I think that provides a cleaner, and clearer API